idem_provider_azurerm/states/azurerm/resource/management_lock.py

```diff
@@ -729,3 +757,202 @@
+
+    '''
+    ret = {
+        'name': name,
+        'result': False,
+        'comment': '',
+        'changes': {}
+    }
+
+    if not isinstance(connection_auth, dict):
+        ret['comment'] = 'Connection information must be specified via connection_auth dictionary!'
+        return ret
+
+    lock = await hub.exec.azurerm.resource.management_lock.get_at_subscription_level(
+        name,
+        azurearm_log_level='info',
+        **connection_auth
+    )
+
+    if 'error' not in lock:
+        if lock_level != lock.get('level'):
+            ret['changes']['level'] = {
+                'old': lock.get('level'),
+                'new': lock_level
+            }
+
+        if notes != lock.get('notes'):
+            ret['changes']['notes'] = {
+                'old': lock.get('notes'),
+                'new': notes
+            }
+
+        if owners:
+            new_owners = owners.sort()
+            lock_owners = lock.get('owners', [])
+            old_owners = []
+            # Extracts the application_id string from each dictionary that represents a ManagementLockOwner object
+            for owner in lock_owners:
+                old_owners.append(owner.get('application_id'))
+            old_owners = old_owners.sort()
+            if old_owners != new_owners:
+                ret['changes']['owners'] = {
+                    'old': old_owners,
+                    'new': new_owners
+                }
+
+        if not ret['changes']:
+            ret['result'] = True
+            ret['comment'] = 'Management lock {0} is already present.'.format(name)
+            return ret
+
+        if ctx['test']:
+            ret['result'] = None
+            ret['comment'] = 'Management lock {0} would be updated.'.format(name)
+            return ret
+
+    else:
+        ret['changes'] = {
+            'old': {},
+            'new': {
+                'name': name,
+                'lock_level': lock_level,
+            }
+        }
+
+        if owners:
+            ret['changes']['new']['owners'] = owners
+        if notes:
+            ret['changes']['new']['notes'] = notes
+
+    if ctx['test']:
+        ret['comment'] = 'Management lock {0} would be created.'.format(name)
+        ret['result'] = None
+        return ret
+
+    lock_kwargs = kwargs.copy()
+    lock_kwargs.update(connection_auth)
+
+    lock = await hub.exec.azurerm.resource.management_lock.create_or_update_at_subscription_level(
+        name=name,
+        lock_level=lock_level,
+        notes=notes,
+        owners=owners,
+        **lock_kwargs
+    )
+
+    if 'error' not in lock:
+        ret['result'] = True
+        ret['comment'] = 'Management lock {0} has been created.'.format(name)
+        return ret
+
+    ret['comment'] = 'Failed to create management lock {0}! ({1})'.format(name, lock.get('error'))
+    if ret['result'] == False:
+        ret['changes'] = {}
+    return ret
+
+
+async def absent_at_subscription_level(hub, ctx, name, connection_auth=None):
```

Comment:
can we roll the resource and subscription level states into a generic "absent" state? maybe handle a parameter that governs where the lock is located.